test/definition_name_test.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:rohd/rohd.dart';
+import 'package:rohd/src/exceptions/name_exceptions.dart';
+import 'package:test/test.dart';
+
+class DefinitionName {
+  String? name;
+  bool isReserved;
+
+  DefinitionName({required this.name, required this.isReserved});
+
+  static String getValidName() => 'specialName';
+  static String getInvalidName() => '/--**definitionName+';
+}
+
+class ValidDefNameModule extends Module {
+  ValidDefNameModule(Logic a, DefinitionName defName)
+      : super(
+          name: 'specialNameInstance',
+          reserveName: false,
+          definitionName: defName.name,
+          reserveDefinitionName: defName.isReserved,
+        ) {
+    addInput('a', a, width: a.width);
+  }
+}
+
+// ignore: camel_case_types
+class byte extends Module {
+  byte(Logic a) {
+    addInput('a', a, width: a.width);
+  }
+}
+
+void main() {
+  group('GIVEN that Reserved Definition Name is TRUE,', () {
+    test('WHEN definition name is valid, THEN expect to compile successfully.',
+        () async {
+      final defName =
+          DefinitionName(name: DefinitionName.getValidName(), isReserved: true);
+      final mod = ValidDefNameModule(Logic(), defName);
+      await mod.build();
+      final sv = mod.generateSynth();
+      // Then it should return compile successfully
+      expect(sv, contains('module specialName('));
```

Comment:
Just like we did for the `Logic` name tests, we can just check `mod.definitionName` immediately rather than generating the SV and parsing it